src/components/app/App.jsx

```diff
@@ -1,5 +1,71 @@
-import React from 'react';
+import React, { useReducer } from 'react';
+
+const initialState = {
+  color: '#b2ff66',
+  before: [],
+  after: []
+};
+
+const reducer = (state, action) => {
+  switch(action.type) {
+    case 'COLOR_CHANGE': {
+      const before = [...state.before, state.color];
```

Comment:
not proper syntax reducer syntax. 
https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers